src/goto-programs/link_to_library.cpp

```diff
@@ -25,24 +79,27 @@ Author: Daniel Kroening, kroening@kroening.com
 void link_to_library(
   goto_modelt &goto_model,
   message_handlert &message_handler,
-  const std::function<
-    void(const std::set<irep_idt> &, symbol_tablet &, message_handlert &)>
-    &library)
+  const std::function<void(
+    const std::set<irep_idt> &,
+    const symbol_tablet &,
+    symbol_tablet &,
+    message_handlert &)> &library)
 {
   // this needs a fixedpoint, as library functions
   // may depend on other library functions
 
   std::set<irep_idt> added_functions;
+  replace_symbolt::expr_mapt object_type_updates;
```

Comment:
can you explain the role of this new map  with a comment

Reply:
Done.